tests/functional/setups/manu_sneddon_2d.py

```diff
@@ -0,0 +1,436 @@
+        )
+
+        bc_val += u_bound
+
+    return bc_val
+
+
+
+
+class ManuExactSneddon2dSetup:
+    """
+    Class for setting up the analytical solution for the pressurized fracture problem.
+    
+    
+    """
+  
+    def __init__(self, setup):
+        # Initialize private variables from the setup dictionary
+        self.p0 = setup.get("p0")
+        self.theta = setup.get("theta")
+
+        self.a = setup.get("a")
+        self.shear_modulus  = setup.get("material_constants").get("solid").shear_modulus
+        self.poi = setup.get("poi")
+        self.length = setup.get("length")
+        self.height = setup.get("height")
+         
+  
+    def exact_sol_global(self, sd):
+        """
+        Compute the analytical solution for the pressurized fracture problem in question.
+        
+        Parameters:
+            sd: Subdomain for which the analytical solution is to be computed.
+            
+        """
+        
+        n = 1000
+        h = 2 * self.a / n
+        box_faces = sd.get_boundary_faces()
+        u_bc = np.zeros((sd.dim, sd.num_faces))
+
+        center = np.array([self.length / 2, self.height / 2, 0])
+        bem_centers = get_bem_centers(self.a, h, n, self.theta, center)
+        eta = compute_eta(bem_centers, center)
+        
+        
+        u_a = -analytical_displacements(self.a, eta, self.p0, self.shear_modulus, self.poi)
+        u_bc = assign_bem(sd, h / 2, box_faces, self.theta, bem_centers, u_a, self.poi)
+        return u_bc
+    
+
+    def exact_sol_fracture( self,
+    gb: pp.MixedDimensionalGrid) -> tuple:
+        """
+        Compute Sneddon's analytical solution for the pressurized crack
+        problem in question.
+
+        Source: Sneddon Fourier transforms 1951 page 425 eq 92
+
+        Parameter
+            gb: Grid object
+            a: Half fracture length of fracture
+            eta: Distance from fracture centre
+            p0: Internal constant pressure inside the fracture
+            G: Shear modulus
+            poi: Poisson ratio
+            height,length: Height and length of domain
+
+        Return:
+            A tuple containing two vectors: a list of distances from the fracture center to each fracture coordinate, and the corresponding analytical apertures.
+        """
+        ambient_dim = gb.dim_max()
+        g_1 = gb.subdomains(dim=ambient_dim - 1)[0]
+        fracture_center = np.array([self.length / 2, self.height / 2, 0])
+
+        fracture_faces = g_1.cell_centers
+
+        # compute distances from fracture centre with its corresponding apertures
+        eta = compute_eta(fracture_faces, fracture_center)
+        apertures = analytical_displacements(self.a, eta, self.p0, self.shear_modulus, self.poi)
+
+        return eta, apertures
+
+
+
+
+
+class ModifiedGeometry(PorePyModel):
+    def set_domain(self) -> None:
+        """Defining a two-dimensional square domain with sidelengths length and height."""
```

Comment:
If it's square, we only need one parameter. We can reuse a class from applications

Reply:
Switched to 'SquareDomainOrthogonalFractures', but I had to overwrite the fractures to fit the setup.